.github/workflows/continuous-integration-terraform.yml

```diff
@@ -0,0 +1,63 @@
+name: Continuous integration
+
+on:
+  push:
+    branches: main
+  pull_request:
+
+jobs:
+  terraform-validate:
+    name: Terraform Validate
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+
+      - name: Check for terraform version mismatch
+        run: |
+          DOTFILE_VERSION=$(cat terraform/.terraform-version)
```

Comment:
Is it also worth adding a terraform linting step?
No worries if the answer is no perhaps it's overkill.